### ISSUE TITLE
[Rule] GROUP rules

### ISSUE THREAD

**Issue by bhirsz** [MarketSquare/robotframework-robocop#1159]
Robot Framework 7.2 introduce new syntax: GROUP. We need to add rules that will cover it.

Proposals:

1. GROUP without name
```
GROUP
    Keyword Call
END
```

2. Nested GROUPs

```
GROUP    Named
    GROUP  At some point we may recommend not using nested groups / use custom keywords instead
        Keyword Call
    END
END
```

3. Too many parameters - already handled by parsing error

```
    GROUP    Too    many    values
        Fail    Not run
    END
```

4. Empty group

```
GROUP    Empty group
END
```

5. Invalid syntax (for some reason parsing error does not catch it):
````
GROUP    Group without end
```

**Comment by bhirsz** [comment]
Also #1158

**Comment by Lakitna** [comment]
How about GROUP with a single keyword in the body?

```robotframework
GROUP  foo
    Do only one thing
END
```

**Comment by Lakitna** [comment]
And maybe a rule for too deeply nested stuff. You can do this without groups, but groups make it easier to create deep nesting:

```robotframework
GROUP  foo
  IF  something
    FOR  i  IN  range(10)
      GROUP  bar
        ...
      END
    END
  END
END
```

**Comment by jannek76** [comment]
any ideas to get this published?

**Comment by bhirsz** [comment]
Took me some time (I apologise to everyone in this thread) - I will make sure to cover rules & see if we have possible existing issues. Some issues could be fixed since last year but I need to check. I'm aiming for upcoming 9.0 release (which should go in a week or two).
